src/rust/src/contract.rs

```diff
@@ -184,6 +185,28 @@ pub fn get_xdr_hashes() -> Vec<XDRFileHash> {
         .collect()
 }
 
+pub const fn get_max_proto() -> u32 {
+    meta::get_ledger_protocol_version(VERSION.interface)
+}
+
+pub fn get_soroban_version_info(core_max_proto: u32) -> SorobanVersionInfo {
+    use meta::{get_ledger_protocol_version, get_pre_release_version};
+    SorobanVersionInfo {
+        env_max_proto: get_ledger_protocol_version(VERSION.interface),
+        env_pkg_ver: VERSION.pkg.to_string(),
+        env_git_rev: VERSION.rev.to_string(),
+        env_pre_release_ver: get_pre_release_version(VERSION.interface),
+        xdr_pkg_ver: VERSION.xdr.pkg.to_string(),
+        xdr_git_rev: VERSION.xdr.rev.to_string(),
+        xdr_base_git_rev: match VERSION.xdr.xdr {
```

Comment:
What does curr/next xdr mean in the context of a given host version? Is next only meaningful for the 'latest' env?

Reply:
No, it's theoretically meaningful on any host -- it selects between the two possible named XDR modules it has available when it imports a given ref of the XDR -- but in practice it'll only _be_ "next" in the context of the latest env, since we route the `core-vnext` feature to only one of the host features `soroban-env-host-p22/next` in Cargo.toml

Reply:
Shouldn't the old envs be using the old XDR as well? Or is this just an implementation artifact where every env exports a different value depending on the feature, but still uses whatever has been linked to it as 'curr'?

Reply:
each env package specifies its own xdr package. they are not unified. it's just that each xdr package also has, inside itself, two modules called `curr` and `prev` that were baked in at the time of packaging.

Reply:
Right, but as far as I understand currently `next` XDR feature can only be enabled for p22 env and generally should only enabled for the 'latest' env that Core supports. That's basically I was referring to in my first comment - it seems to me like 'next' version should only be set in the latest env. If that sounds correct to you, maybe we should make this an assertion? I just want to be sure we don't accidentally use 'next' XDR of the old env and then spend hours figuring out potential catchup failure reason.

Reply:
Right and if you look in Cargo.toml you'll see the line `core-vnext = ["soroban-env-host-p22/next"]`

This means that enabling core-vnext will only enable the next feature on the p22 host, not any other hosts.

I can try adding an assertion if you like; it'll just be checking that this line exists in the cargo file, but I don't mind more cross checks!

Reply:
> I can try adding an assertion if you like; it'll just be checking that this line exists in the cargo file, but I don't mind more cross checks!

I think it's pretty easy to forget to update this line among all the protocol migration upgrades, so the assertion would be much appreciated, thanks!